flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/filesystem/FileSystemOptions.java

```diff
@@ -88,4 +88,48 @@
 							" If timestamp in partition is year, month, day, hour," +
 							" can configure: '$year-$month-$day $hour:00:00'." +
 							" If timestamp in partition is dt and hour, can configure: '$dt $hour:00:00'.");
+
+	public static final ConfigOption<String> SINK_PARTITION_COMMIT_TRIGGER =
+			key("sink.partition-commit.trigger")
+					.stringType()
+					.defaultValue("partition-time")
+					.withDescription("Trigger type for partition commit:" +
+							" 'partition-time': extract time from partition," +
+							" if 'watermark' > 'partition-time' + 'delay', will commit the partition." +
+							" 'process-time': use processing time, if 'current processing time' > " +
+							"'partition creation time' + 'delay', will commit the partition.");
+
+	public static final ConfigOption<Duration> SINK_PARTITION_COMMIT_DELAY =
+			key("sink.partition-commit.delay")
+					.durationType()
+					.defaultValue(Duration.ofMillis(0))
+					.withDescription("The partition will not commit until the delay time." +
+							" if it is a day partition, should be '1 d'," +
+							" if it is a hour partition, should be '1 h'");
+
+	public static final ConfigOption<String> SINK_PARTITION_COMMIT_POLICY_KIND =
+			key("sink.partition-commit.policy.kind")
+					.stringType()
+					.noDefaultValue()
+					.withDescription("Policy to commit a partition is to notify the downstream" +
+							" application that the partition has finished writing, the partition" +
+							" is ready to be read." +
+							" metastore: add partition to metastore." +
```

Comment:
If users choose metastore policy, don't they need to specify/provide a `TableMetaStoreFactory` implementation?

Reply:
I will add comment: `Only work with hive table, it is empty implementation for file system table.`